test/e2e/examples.go

```diff
@@ -171,6 +276,38 @@ func makeHttpRequestToService(c *client.Client, ns, service, path string) (strin
 	return string(result), err
 }
 
+func getNsCmdFlag(ns *api.Namespace) string {
+	return fmt.Sprintf("--namespace=%v", ns.Name)
+}
+
+// pass enough context with the 'old' parameter so that it replaces what your really intended.
+func prepareResourceFileWithReplacedString(inputFile, old, new string) string {
```

Comment:
maybe instead of creating temporary files write updated defintion to kubectl's stdin. 
See https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/util.go#L833
and https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/kubectl_create.md.